app/src/main/java/org/fossasia/openevent/app/common/data/models/CustomObjectWrapper.java

```diff
@@ -0,0 +1,15 @@
+package org.fossasia.openevent.app.common.data.models;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Created by priyanshu on 8/1/18.
+ */
+
+public final class CustomObjectWrapper {
+
+    public static <E> Map<String, E> withLabel(String outerWrap, E wrappedObject) {
+        return Collections.singletonMap(outerWrap, wrappedObject);
+    }
+}
```

Comment:
Please remove author comments

Reply:
Also, why is this used?

Reply:
The token Json request has only email in Json body wrapped by defauly root element data. So to wrap the RequestToken POJO by data element i have to use a custom wrapper otherwise it wasn't getting wrapped. I couldn't use SerializationFeature.WRAP_ROOT_ELEMENT on ObjectWrapper because there was only one instance of it and it was mapped to all the mapped classses ie changing it would have effect on all the mapped classes.

Reply:
That's not the reason actually. You have to use `jsonapi-converter` for the model. See other models like event.

Reply:
Using josnapi-converter, 3 tags come by default - type, id, attributes and the body gets wrapped by attributes tag. I've tried to modify it but can't get it to the required form - 
{"data":{"type":"token","id":"0","attributes":{"email":"foxvixen@fox.com"}}}

I've come up with 2 ways of doing this - 
1. With CustomObjectWrapper I'm able to wrap RequestToken object with data tag and serialise it. 2. By extending the ObjectMapper class and changing it to wrap those objects with the root element tag ie. "data" which have a particular annotation on them like JsonRootName("data") but i think first one's better.

Reply:
OK @Masquerade0097 seems like the problem is that API does not conform to JSONAPI always. Can you please add a TODO message in the API interface to revert to JSONAPI implementation once this is fixed. Also please add a comment on CustomDataWrapper about why it is used.

The implementation of your PR is of extremely good quality, thanks a lot :smile:

Reply:
Ya, I'll add the comment and TODO. Also I've written some tests for the files which i have made or modified so should I squash them in this commit or make a separate commit for that.

Reply:
Both ways are fine